shared/middleware/assets_api.rb

```diff
@@ -59,16 +31,10 @@ def self.s3
   #
   get %r{/v3/assets/([^/]+)/([^/]+)$} do |encrypted_channel_id, filename|
     dont_cache
-    not_found unless type = filename.split('.').last
+    not_found unless (type = filename.split('.').last)
```

Comment:
I believe this won't have the intended effect if the filename doesn't include a ".".  For example:
"readme".split('.').last
=> "readme"